File: Projects/Project_One/projectone_01.py
"""
Name: Badr Choubai
Professor: David Kramer
Class: CS 1030
Project: Python Project One
"""


def calculate_meters(inches: int) -> float:
    inches_to_centimeters = (lambda inches: inches * 2.54)
    centimeters_to_meters = (lambda centimeters: centimeters / 100)
    return (centimeters_to_meters(inches_to_centimeters(inches)))


def main():
    height_input_feet = input('Feet: ')
    height_input_inches = input('Inches: ')
    height_str = f"{height_input_feet}'{height_input_inches}\""
    output_str = ""

    feet = int(height_input_feet)
    inches = int(height_input_inches)
    inches_total = (feet * 12) + inches

    if inches_total <= 95:
        meters = calculate_meters(inches_total)
        output_str = f"Original Height Input {height_str} -> Converted to meters {meters:.2f}m"
    else:
        output_str = "Wow, you're so tall!"

    print(output_str)


if __name__ == "__main__":
    main()

File: Projects/Project_One/Java/ProjectOne01.java
package Projects.Project_One.Java;

import java.util.Scanner;

class ProjectOne01 {
    private static Scanner inputScanner = new Scanner(System.in);

    private static String getUserInput(String prompt) {
        System.out.printf("%s:\n> ", prompt);
        var input = inputScanner.nextLine();
        return input;
    }

    public static void main(String[] args) {
        String feetInput = getUserInput("Feet");
        String inchesInput = getUserInput("Inches");
        var heightString = String.format("%s'%s\"", feetInput, inchesInput);

        var feet = Integer.parseInt(feetInput);
        var inches = Integer.parseInt(inchesInput);

        var total = (feet * 12) + inches;

        if (total >= 95) {
            System.out.println("Wow! You're so tall");
        } else {
            var centimeters = total * 2.54;
            var meters = centimeters / 100;

            System.out.printf("Original height input %s, to meters: %.2fm", heightString, meters);
        }
    }
}
